### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by TotalDwarf03**
Changes involve:

- Modifying the code to use a python package to interact with the GH API instead of stored local version
- Changing any background variables to come from the environment instead of being written in code
- Small bug fixes
- Add Secret Scanning and Dependabot checks

### CODE REVIEW

**Review by gibbardsteve — COMMENTED** [review]
Some code scanning warnings that need to be addressed (it looks like there is a concern the json might contain sensitive info and be stored in plain text) I am not familiar enough with the response to know.

File: aws_lambda/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,16 @@
+boto3==1.34.123
+botocore==1.34.145
+certifi==2024.7.4
+cffi==1.16.0
+charset-normalizer==3.3.2
+cryptography==43.0.0
+github-api-toolkit @ git+https://github.com/ONS-Innovation/github-api-package.git
```
Comment (gibbardsteve) [inline aws_lambda/requirements.txt:7]: Will want to pin to a release (separate JIRA to cover this)

**Review by gibbardsteve — APPROVED** [review]

File: aws_lambda/requirements.txt (hunk 1 above)
Comment (TotalDwarf03) [reply aws_lambda/requirements.txt:7]: Jira ticket added, will amend next sprint

**Comment by TotalDwarf03** [comment]
Discussed and dismissed code scanning alerts w/ Steve Gibbard. All stored JSON responses are kept while the container is running, and lost once they're destroyed. This means there is minimal risk. 

Added ticket to Jira to add a future enhancement to write directly to S3 instead of locally then uploading it.

**Pull request merged**